src/main/java/net/snowflake/client/core/SFResultSetFactory.java

```diff
@@ -24,14 +24,17 @@
    */
   static SFBaseResultSet getResultSet(JsonNode result, SFStatement statement, boolean sortResult)
       throws SQLException {
+
+    // This should only be invoked from an SFSession connection
+    SFSession session = (SFSession) statement.getSession();
```

Comment:
SFStatement.getSession should return SFSession and not SFSessionInterface, correct ?